common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/challengehandlers/NtlmChallengeHandler.java

```diff
@@ -22,6 +22,95 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.ui.embeddedwebview.challengehandlers;
 
-final class NtlmChallengeHandler {
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.EditText;
 
+import com.microsoft.identity.common.R;
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+/**
+ * Http authorization handler for NTLM challenge on web view.
+ */
+public final class NtlmChallengeHandler implements IChallengeHandler<NtlmChallenge, Void> {
+    private static final String TAG = NtlmChallengeHandler.class.getSimpleName();
+    private Activity mActivity;
+    private IChallengeCompletionCallback mChallengeCallback;
+
+    /**
+     * Constructor of NtlmChallengeHandler.
+     * @param activity activity to place the UI
+     * @param callback challenge completion callback which will process the challenge result.
+     */
+    public NtlmChallengeHandler(final Activity activity,
+                                final IChallengeCompletionCallback callback) {
+        mActivity = activity;
+        mChallengeCallback = callback;
+    }
+
+    /**
+     * Process the NTLM Challenge. If the credentials stored for the current host exists, use the
+     * users credentials to resolve the NTML challenge. Otherwise, show the http auth dialog on UI,
+     * user will need to type in the username and password to resolve the NTML challenge.
+     */
+    @Override
+    public Void processChallenge(final NtlmChallenge ntlmChallenge) {
+        if (ntlmChallenge.getHandler().useHttpAuthUsernamePassword()
+                && ntlmChallenge.getView() != null) {
+            final String[] haup = ntlmChallenge.getView()
+                    .getHttpAuthUsernamePassword(ntlmChallenge.getHost(), ntlmChallenge.getRealm());
+            if (haup != null && haup.length == 2) {
+                final String userName = haup[0];
+                final String password = haup[1];
+                if (userName != null && password != null) {
```

Comment:
for the String 'username' and 'password', checking 'null' is not enough, also please check the length of the String